web/client/components/background/__tests__/PreviewIcon-test.jsx

```diff
@@ -36,29 +36,43 @@ describe("test the PreviewIcon", () => {
             setCurrentBackgroundLayer: () => {}
         };
 
-        const layer = {
+        const layers = [{
             type: "empty",
             visibility: true,
             id: 'layer'
-        };
+        },
+        {
+            type: "wms",
+            visibility: true,
+            id: 'layer'
+        },
+        {
+            type: "osm",
+            visibility: true,
+            id: 'layer'
+        }];
 
         const spyPropertiesChange = expect.spyOn(actions, 'onPropertiesChange');
         const spyToggle = expect.spyOn(actions, 'onToggle');
         const spyLayerChange = expect.spyOn(actions, 'onLayerChange');
         const spySetCurrentBackgroundLayer = expect.spyOn(actions, 'setCurrentBackgroundLayer');
 
-        const previewIcon = ReactDOM.render(<PreviewIcon onPropertiesChange={actions.onPropertiesChange} onToggle={actions.onToggle} onLayerChange={actions.onLayerChange} setCurrentBackgroundLayer={actions.setCurrentBackgroundLayer} vertical layer={layer}/>, document.getElementById("container"));
-        expect(previewIcon).toExist();
-        const node = ReactDOM.findDOMNode(previewIcon);
-        expect(node).toExist();
-        const image = node.querySelector('.background-preview-icon-frame').children[0];
-        ReactTestUtils.Simulate.mouseOver(image);
-        expect(spyLayerChange).toHaveBeenCalled();
-        ReactTestUtils.Simulate.click(image);
-        expect(spyPropertiesChange).toHaveBeenCalled();
-        expect(spyToggle).toHaveBeenCalled();
-        expect(spyPropertiesChange).toHaveBeenCalledWith("layer", {visibility: true});
-        expect(spySetCurrentBackgroundLayer).toHaveBeenCalledWith("layer");
+        layers.map((layer, i)=>{
```

Comment:
If you are doing side effects and not collecting the map function result, I prefer to use forEach, it makes the intent more clear, in my opinion